### ISSUE TITLE
response lifetime appears to require 'static, which makes it tricky to pass references.

### ISSUE THREAD

**Issue by mutantbob** [bobbobbio/http_io#1]
I am experimenting with serving camera images from an ESP32 microcontroller.  It has a `std` environment available.  In order to fetch the image, I must repeatedly communicate with the camera via SPI.

I store a reference to the camera in `MyHandler` and pass it to the `CameraBody` (: `impl Read`) , but when I wrap that CameraBody in a `Box` and an `HttpResponse`, I get

```
error[E0495]: cannot infer an appropriate lifetime for lifetime parameter `'a` due to conflicting requirements
  --> src/httpd.rs:45:24
   |
45 |             let body = CameraBody::new(self.camera, self.spi).map_err(|e| {
   |                        ^^^^^^^^^^^^^^^
   |
note: first, the lifetime cannot outlive the lifetime `'a` as defined here...
  --> src/httpd.rs:21:6
   |
21 | impl<'a, SPI, CS, I2C, ES, EI, S> HttpRequestHandler<S> for MyHandler<'a, SPI, CS, I2C, ES, EI>
   |      ^^
note: ...so that reference does not outlive borrowed content
  --> src/httpd.rs:45:40
   |
45 |             let body = CameraBody::new(self.camera, self.spi).map_err(|e| {
   |                                        ^^^^^^^^^^^
   = note: but, the lifetime must be valid for the static lifetime...
note: ...so that the types are compatible
  --> src/httpd.rs:66:9
   |
66 |         Ok(resp)
   |         ^^^^^^^^
   = note: expected `Result<HttpResponse<Box<(dyn std::io::Read + 'static)>>, HttpResponse<Box<(dyn std::io::Read + 'static)>>>`
              found `Result<HttpResponse<Box<dyn std::io::Read>>, HttpResponse<Box<dyn std::io::Read>>>`
```

I get the feeling I'm going to have to wrap some things in an `Arc<Mutex<_>>` to help them live longer.  This will get significantly harder if I target a no_std environment (which often lack `Arc` or `Mutex`).

I can't help but wonder if the API could be upgraded to have a lifetime on the result of the `get()` function to simplify things.

**Comment by bobbobbio** [comment]
Yeah, I think the trait definition for the handler has a silent `'static` in the boxes. That should be easy to fix, I'll try to make it more general in that way.

**Comment by bobbobbio** [comment]
fixed in 9ba98e8fa4f9e58da3437b2a2054e314b54dbee4
